File: dsa in python/binary search.py
arr = list(range(1, 21))
target = int(input("Enter target you want to find: "))
start = 0
end = len(arr) - 1
found = False

while start <= end:
    middle = (start + end) // 2
    if arr[middle] == target:
        print(f"{target} found at index {middle}")
        found = True
        break
    elif arr[middle] > target:
        end = middle - 1
    else:
        start = middle + 1

if not found:
    print(f"{target} not found in arr 1-20")

File: dsa in c/binarysearch.c
#include <stdio.h>
int main(){
    int N = 20, arr[N], target;
    printf("Enter %d numbers: ", N);
    for(int i = 0; i < N; i++) scanf("%d", &arr[i]);
    printf("Enter number to find: ");
    scanf("%d", &target);
    int low = 0, high = N - 1;
    while(low <= high){
        int mid = (low + high) / 2;
        if(arr[mid] == target){
            printf("%d found at index %d.\n", target, mid);
            return 0;
        }
    }
    printf("%d not found in given array.\n", target);

    return 0;
